### PR TITLE
[MM-56387] Replace missed usage of LocalizedIcon in 'search_results_header.tsx' with i/span tags

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
- Made all back buttons in RHS consistent
- Brought the aria labels to the buttons instead of icons
- Added translations to the button's aria labels where it was missing
- converted search_result_header component to functional component
- Some back buttons where of link tag, unified them to use the same buttons

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56387

#### Screenshots
Screenshot 2024-01-01 at 13 06 27

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/sass/layout/_sidebar-right.scss [hunk 1]
```diff
@@ -237,6 +213,10 @@
             font-size: 20px;
         }
 
+        .sidebar--right__back {
+            margin-right: 4px !important;
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/sass/layout/_sidebar-right.scss:217]: This is required because of the button class have margin as 0 with important in them,

**Review by asaadmahmood — APPROVED** [review]
Looks great to me!

**Review by yasserfaraazkhan — APPROVED** [review]

**Review by fmartingr — APPROVED** [review]

**Pull request merged**